compiler/pragmas.nim

```diff
@@ -567,6 +567,9 @@ proc semAsmOrEmit*(con: PContext, n: PNode, marker: char): PNode =
         var e = searchInScopes(con, getIdent(con.cache, sub), amb)
         # XXX what to do here if 'amb' is true?
         if e != nil:
+          if e.kind in routineKinds:
+            localError(con.config, n.info, "symbols cannot be routine kinds")
+            return
```

Comment:
Non-overloaded procs should be allowed -- you would get the proc's address in the inline assembler, for example.